front/pages/api/stripe/webhook.ts

```diff
@@ -277,10 +283,60 @@ async function handler(
             { event },
             "[Stripe Webhook] Received customer.subscription.updated event."
           );
+          const previousAttributes = event.data.previous_attributes;
+          if (!previousAttributes) break; // should not happen by definition of the subscription.updated event
+
+          if (
+            // The subscription is canceled (but not yet ended) or reactivated
+            stripeSubscription.status === "active" &&
+            "cancel_at_period_end" in previousAttributes
+          ) {
+            // first update subscription endDate in our database
+            // cancel_at set means the user just canceled, unset means the user just reactivated
+            const endDate = stripeSubscription.cancel_at
+              ? new Date(stripeSubscription.cancel_at * 1000)
+              : null;
+
+            // get subscription
+            const subscription = await Subscription.findOne({
+              where: { stripeSubscriptionId: stripeSubscription.id },
+            });
+            if (!subscription) {
+              return apiError(req, res, {
+                status_code: 500,
+                api_error: {
+                  type: "internal_server_error",
+                  message:
+                    "Stripe Webhook: canceling subscription: Subscription not found.",
+                },
+              });
+            }
+            await subscription.update({ endDate });
+
+            // then email admins
+            const adminEmails = await getAdminEmailsForWorkspace(
+              subscription.workspaceId
+            );
+            if (adminEmails.length === 0) {
+              return apiError(req, res, {
+                status_code: 500,
+                api_error: {
+                  type: "internal_server_error",
+                  message:
+                    "Stripe Webhook: canceling subscription: Error getting admin emails.",
+                },
+              });
+            }
+            // send email to admins
+            for (const adminEmail of adminEmails)
+              await sendEmail(
```

Comment:
Could we rework this and not export `sendEmail` but instead have something like: 

```
endDate? sendSubscriptionCancelMessage(endDate) : sendSubscriptionReactivateMessage() 
```

With `sendSubscriptionCancelMessage` & `sendSubscriptionReactivateMessage` directly sending the email 🙏🏻